include/envoy/upstream/upstream.h

```diff
@@ -699,6 +700,11 @@ class ClusterInfo {
    */
   virtual std::chrono::milliseconds connectTimeout() const PURE;
 
+  /**
+   * @return maximum duration time to keep alive stream
+   */
+  virtual absl::optional<std::chrono::milliseconds> maxStreamDuration() const PURE;
```

Comment:
Sorry I should have noticed this before but this should be in one of the HTTP settings message below (or a new common one) given this is HTTP specific.